docs/4.2/features/enhanced_session_recording.md

```diff
@@ -0,0 +1,185 @@
+
+The disadvantages is that session recordings can by bypassed using several techniques.
+
+- **Obfuscation**. For example, even though the command ` echo Y3VybCBodHRwOi8vd3d3LmV4YW1wbGUuY29tCg== | base64 --decode | sh` does not contain 
+`curl http://www.example.com`, when decoded, that is what is run.
+- **Shell scripts**. For example if a user uploads and executes a script, the commands 
+run within the script are not captured, simply the output.
+- **Terminal controls**. Terminals support a wide variety of controls including the 
+ability for users to disable terminal echo. This is frequently used when requesting
+ credentials. Disabling terminal echo allows commands to be run without being captured.
+
+Furthermore, due to their unstructured nature, session recordings are difficult to 
+ingest and perform monitoring/alerting on.
+
+!!! Note:
+
+    Enhanced Session Recording requires all parts of the Teleport system to be running
+    4.2+. 
+
+# Requirements:
+
+## 1. Check / Patch Kernel.
+Teleport 4.2 with Enhanced Session Recording requires Linux kernel 4.18 (or above) as 
+well as kernel headers. 
+
+You can check your kernel version using the `uname` command. The output should look 
+something like the following.
+
+```
+$ uname -a
+Linux ip-172-31-43-104.ec2.internal 4.19.72-25.58.amzn2.x86_64 x86_64 x86_64 x86_64 GNU/Linux
+```
+
+
+### Ubuntu 
+
+|       |                   | Kernel Version        |
+|-------|-------------------|-----------------------|
+| 18.10 | Cosmic Cuttlefish | 4.18 [Patch Kernel](http://www.theubuntumaniac.com/2018/11/update-install-kernel-4191-stable-on.html)  |
+| 19.04 | Disco Dingo       | 5.0 ✅                  |
+| 19.10 | Eoan Ermine       | 5.3 ✅                  |
+
+### Debian
+
+|     |                     | Kernel Version            |
+|-----|---------------------|---------------------------|
+| 9   | Debian Stretch      | 4.9.0-6 [Patch Kernel](https://wiki.debian.org/HowToUpgradeKernel) |
+| 10  | Buster              | 4.19 ✅                   |
+
+### CentOS
+|               | Kernel Version         |
+|---------------|------------------------|
+| 8.0-1905	    |            4.18.0.80 ✅  |
+
+### Red Hat 
+|                     | Kernel Version         |
+|---------------------|------------------------|
+| Enterprise Linux 8  |          4.18.0-147 ✅ |
+
+### Amazon Linux 
+We recommend using `Amazon Linux 2` to install and use Linux kernel 4.19 using
+`sudo amazon-linux-extras` install kernel-ng and rebooting your instance. 
+
+### archlinux 
+|                     | Kernel Version         |
+|---------------------|------------------------|
+| 2019.12.01          |              5.3.13 ✅ |
+
+## 2. Install BCC Tools 
+
+Run the following script to download the prerequisites to build BCC tools, building LLVM and Clang targeting BPF byte code, and then building and installing BCC tools.
+
+!!! note
+    We plan have plans to remove this step after we've moved our of our beta.
```

Comment:
We plan to soon support installing `bcc-tools` from packages instead of compiling them yourself to make taking advantage of enhanced session recording easier.

Reply:
also a note that this script is just for CentOS

Reply:
https://github.com/iovisor/bcc/blob/master/INSTALL.md#debian---source <-- Suggest linking to this reference